### PR TITLE
[public-api] Setup handler for charge.dispute.created

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Initial setup for handling of stripe disputes. See https://stripe.com/docs/api/events/types for details

We haven't yet configured Stripe to send these events in our webhook config, but we will enable it for preview/staging to validate we're correctly picking up the events (and returning unimplemented)

## Related Issue(s)

* Part of https://github.com/gitpod-io/ops/issues/8192
* Depends on https://github.com/gitpod-io/gitpod/pull/17036

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/public-api-server/pkg/webhooks/stripe.go [hunk 1]
```diff
@@ -59,30 +59,46 @@ func (h *webhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	switch event.Type {
 	case "invoice.finalized":
-		invoiceId, ok := event.Data.Object["id"].(string)
+		invoiceID, ok := event.Data.Object["id"].(string)
```
Comment (easyCZ) [inline components/public-api-server/pkg/webhooks/stripe.go:62]: Renames to conform to Go style, where shorthands are capitalized fully

File: components/public-api-server/pkg/webhooks/stripe.go [hunk 2]
```diff
@@ -59,30 +59,46 @@ func (h *webhookHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 	case "customer.subscription.deleted":
-		subscriptionId, ok := event.Data.Object["id"].(string)
+		subscriptionID, ok := event.Data.Object["id"].(string)
 		if !ok {
 			log.Error("failed to find subscriptionId id in Stripe event payload")
 			w.WriteHeader(http.StatusBadRequest)
+			return
```
Comment (easyCZ) [inline components/public-api-server/pkg/webhooks/stripe.go:80]: Was missing a return here

File: components/public-api-server/pkg/webhooks/stripe_test.go [hunk 3]
```diff
@@ -27,6 +27,7 @@ const (
 	invoiceFinalizedEventType   = "invoice.finalized"
 	customerCreatedEventType    = "customer.created"
 	customerSubscriptionDeleted = "customer.subscription.deleted"
+	chargeDisputeCreated        = "charge.dispute.created"
```
Comment (geropl) [inline components/public-api-server/pkg/webhooks/stripe_test.go:30]: Why do we the constants here and are not using them in the impl? Just asking if that is intentional (could imagine).

**Review by geropl — APPROVED** [review]
Code LGTM

File: components/public-api-server/pkg/webhooks/stripe_test.go (hunk 3 above)
Comment (easyCZ) [reply components/public-api-server/pkg/webhooks/stripe_test.go:30]: Yeah makes sense to use them in the handler switch also. I'll raise a follow-up.

**Pull request merged**